### PR TITLE
added main + embeddings + linted

### PR DESCRIPTION
**Pull request by Jonah-gr**
1. Added main file to get metadata and similarities via the console: with, e.g. `python -m resources.main -m -s -p "D:/data/image_data" --checkpoint 10` the program will run through every image and saves the metadata and similarities. The similarities will be saved in a pickle file every 10 images. These are all arguments right now: 
image

2. Added embeddings by caluclating the feature vectors with the InceptionV3 net. The similarities are saved in a dictionary, e.g. 
```python 
{1: [],
2: []}
```
with the image ids as keys and the similarity measures as a list. We can discuss ich we want to keep working with dictionaries here or a Dataframe for example.
3. Added gitignore. Database and pkl wont be saved in repo.
4. linted everything with black.

I hope you also think thats a good idea to work with argparse here. If we now add 2 more similarity measure methods, we can run through this whole thing.
If you have any questions, please ask:)

### CODE REVIEW

File: resources/similarity.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from resources.embedding import inception_v3
+
+### import other similarity measure methods
```
Comment (AlhaririAnas) [inline resources/similarity.py:3]: Instead of importing the three methods for calculating similarities from a separate Python file, we could implement them directly in `similarity.py`. This would help keep the number of Python files manageable, especially since it's just a few lines of code. Do you have a different opinion on this? If you agree, I could make the changes on the [color_vector_and_similarity](https://github.com/AlhaririAnas/Image-recommender/tree/color_vector_and_similarity) branch.

File: resources/embedding.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import torch
+import torch.nn as nn
+from torchvision import models, transforms
+from torchvision.models.inception import Inception_V3_Weights
+
+
+def prepocess(img):
+    try:
+        transform = transforms.Compose([transforms.Resize(299), transforms.CenterCrop(299), transforms.ToTensor()])
```
Comment (AlhaririAnas) [inline resources/embedding.py:9]: Suppose we have an image that is 500x400 pixels:

- Applying transforms.Resize(299) scales the image to about 374x299 pixels.
- Then, transforms.CenterCrop(299) takes a centered 299x299 pixel portion from it.

Doesn't this cause a loss of information that could be critical?

Also, why was the size of 299x299 pixels chosen? Is this a specific requirement for the Inception_V3 model?

File: resources/similarity.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+
+def get_similarities(img, args):
```
Comment (AlhaririAnas) [inline resources/similarity.py:6]:
```suggestion
def get_similarities_vectors(img, args):
```
Maybe this name is clearer?

At the end, the function should return two lists: one containing the embedding vector and the other containing the color vector. Both lists should be inserted into a pickle file along with the corresponding image ID at the same time: return [[embedding_vector], [color_vector]]. What do you think about it?

**Review by AlhaririAnas — COMMENTED** [review]
:)

**Review by AlhaririAnas — APPROVED** [review]
